crates/e2k-py/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+use pyo3::prelude::*;
+use pyo3::types::{PyBytes, PyDict, PyList, PyType};
+
+fn extract_strategy(strategy: &str, kwargs: Option<&Bound<'_, PyDict>>) -> PyResult<e2k::Strategy> {
+    Ok(match strategy {
+        "greedy" => e2k::Strategy::Greedy,
+        "top_k" => {
+            let k = kwargs.and_then(|kwargs| {
+                kwargs
+                    .get_item("k")
+                    .ok()
+                    .flatten()
+                    .map(|k| k.extract::<usize>())
+            });
+            match k {
+                Some(Err(err)) => return Err(err),
+                Some(Ok(k)) => e2k::Strategy::TopK(e2k::StrategyTopK { k }),
+                None => e2k::Strategy::TopK(e2k::StrategyTopK::default()),
+            }
+        }
```

Comment:
見た目`Err`の握り潰しが起きてそうに見えるのでちょっと抵抗感を覚えてしまいました。

ちょっと雑ですがこんな感じでいけそう。

```suggestion
        "top_k" => e2k::Strategy::TopK(
            kwargs
                .and_then(|kwargs| kwargs.get_item("k").transpose())
                .map(|k| k.and_then(|k| k.extract()))
                .transpose()?
                .map(|k| e2k::StrategyTopK { k })
                .unwrap_or_default(),
        ),
```